### ISSUE TITLE
Rider clicking cancel on carpool the driver has cancelled returns 404

### ISSUE THREAD

**Issue by simonl2002** [RagtagOpen/nomad#199]
**Description**
A user clicking cancel on a carpool which the driver has just cancelled results in a 404 error page.

**Steps**
1. Rider requests a ride
2. Driver cancels ride
3. Rider clicks cancel 

**Actual Results**
The rider gets a 404 page.

**Expected Results**
The rider should be told that the rider was cancelled.

**Comment by simonl2002** [comment]
I think for drivers/riders if they follow email links etc the pages of canceled ride should probably still render even if they can't access these rides via the UI.

**Comment by iandees** [comment]
This happens because we delete the ride request from the database when its canceled. We could keep the ride request around in a canceled state, but what actions would a user take from that point? Re-request?

**Comment by simonl2002** [comment]
@iandees They wouldn't need to necessarily need to be able to take any action. It is just a bit jarring to hit a 404 when there was something there before.

**Comment by jillh510** [comment]
Agreed that "we can't find this carpool" or similar would be preferable to a 404.

**Comment by kdechant** [comment]
In my experience (e.g., on Meetup), there are some users who tend to just delete things without notifying anyone that they're doing it. For this reason, it would be nice to have a "cancelled" status with a message, in case a rider doesn't see the email notification that it's been cancelled.

I think it would also be good to have the cancelled carpools appear on the rider's "My Rides" page (with a visual indicator like strike-through) to give the riders another indication that it was cancelled.

I'm not suggesting that there be any action available once a carpool has been cancelled; just information about what happened.

**Comment by kdechant** [comment]
I'm happy to work on this but not until 9/4, so if anyone wants to tackle it sooner, go ahead.

**Comment by jillh510** [comment]
@kdechant good ideas about "cancelled" status. I'll file a separate issue.
